support/github-to-jira-synchronization/clay.md

```diff
@@ -0,0 +1,257 @@
+    ,"regression"  : "Bug"
+    ,"chore"       : "Task"
+    ,"enhancement" : "Story"
+]
+
+issue.projectKey   = "IFI"
+
+def labelType = replica.labels.find { it.label.startsWith("type:") }?.label
+def typeMapped = labelTypeMapping[labelType?.replace("type:", "")?.trim()?.toLowerCase()]
+issue.typeName     = nodeHelper.getIssueType(typeMapped, issue.projectKey)?.name ?:
+                         (nodeHelper.getDefaultIssueType(issue.projectKey)?.name ?: "Task")
+
+issue.assignee     = findUser(replica.assignees[0])
+issue.reporter     = userHelper.getByUsername("exalate")
+issue.summary      = replica.summary
+issue.description  = replica.description
+issue.labels       = replica.labels.collect { it.label = it.label.replace(" ", "_"); it }
+issue.comments     = commentHelper.mergeComments(issue, replica,
+                         { comment ->
+                             def userAuthor = findUser(comment.author)
+                             if (userAuthor.username == "team-frontend-infra")
+                                 comment.body = comment.author.displayName + " commented: \n" +
+                                     "{quote}\n" +
+                                     comment.body + "\n" +
+                                     "{quote}\n"
+                             else
+                                 comment.executor = userAuthor
+                          })
+
+def milestoneIssueKey = replica.fixVersions[0]?.name =~ /[\w]+-[\d]+/
+def issueEpic = ComponentAccessor.getIssueManager().getIssueObject(milestoneIssueKey ? milestoneIssueKey[0] : "")
+issue.customFields."Epic Link".value = issueEpic?.getIssueType()?.getName() == "Epic" ? issueEpic : null
+
+final def transitionsMapping = [
+    transitions: [
+        [ name:'Exl to "In Progress"'  , to:'In Progress']
+    ]
+]
+
+def targetStatus = null
+if (replica.assignees.size() > 0)
+    targetStatus = "In Progress"
+
+def transition = transitionsMapping.transitions.find { it.to == targetStatus }
+if (transition)
+    workflowHelper.transition(issue, transition.name)
+
+
+// --- functions --- //
+def findUser(replicaUser) {
+    if (!replicaUser)
+        return null
+
+    def user = userHelper.getByUsername(replicaUser.username)
+
+    if (user && user.active && userHelper.isGroupMember(replicaUser.username, "liferay-staff"))
+        return user
+
+    final def usernameMapping = [
+        "ambrinchaudhary": "ambrin.chaudhary"
+        ,"brunofernandezg": "bruno.fernandez"
+        ,"bryceosterhaus": "bryce.osterhaus"
+        ,"carloslancha": "carlos.lancha"
+        ,"diegonvs": "diego.nascimento"
+        ,"dsanz": "daniel.sanz"
+        ,"eduardoallegrini": "Eduardo.allegrini"
+        ,"izaera": "ivan.zaera"
+        ,"jbalsas": "jose.balsas"
+        ,"jonmak08": "jonathan.mak"
+        ,"julien": "julien.castelain"
+        ,"marcoscv-work": "marcos.castro"
+        ,"matuzalemsteles": "matuzalem.teles"
+        ,"pat270": "patrick.yeo"
+        ,"wincent": "greg.hurrell"
```

Comment:
And you didn't say anything about this one, @nhpatt 😢

Reply:
Yeah, I was going to say that this could be a dictionary at the cost of performance.

Reply:
In any case, this is the current script in JIRA... after this, you're free too make any changes you want!

![yolo](https://user-images.githubusercontent.com/905006/118645246-95db6580-b7d6-11eb-8007-a7d33f6a8f7a.gif)

Reply:
![trap](https://user-images.githubusercontent.com/74542632/118653474-da233180-b7e7-11eb-9ebd-4cc8126f9c90.gif)